custom_components/unifiprotect/binary_sensor.py

```diff
@@ -165,12 +174,48 @@ async def async_setup_entry(
 ) -> None:
     """Set up binary sensors for UniFi Protect integration."""
     entry_data: UnifiProtectEntryData = hass.data[DOMAIN][entry.entry_id]
-    entities = _async_device_entities(entry_data)
+    entities: list[UnifiProtectBinarySensor] = _async_device_entities(entry_data)
     entities += _async_nvr_entities(entry_data)
+    entities += _async_motion_entities(hass, entry_data)
 
     async_add_entities(entities)
 
 
+@callback
+def _async_motion_entities(
+    hass: HomeAssistant,
+    entry_data: UnifiProtectEntryData,
+) -> list[UnifiProtectAccessTokenBinarySensor]:
+    protect = entry_data.protect
+    protect_data = entry_data.protect_data
+
+    entities = []
+    for device in protect_data.get_by_types({ModelType.CAMERA}):
+        for description in MOTION_SENSORS:
+            entities.append(
+                UnifiProtectAccessTokenBinarySensor(
+                    protect, protect_data, device, description
+                )
+            )
+            _LOGGER.debug(
+                "Adding binary sensor entity %s for %s",
+                description.name,
+                device.name,
+            )
+
+    @callback
+    def update_tokens(time: datetime) -> None:
+        """Update tokens of the entities."""
+        for entity in entities:
+            entity.async_update_token()
+            entity.async_write_ha_state()
```

Comment:
We aren't allowed to call `async_write_ha_state` from outside the entity. This needs to use `async_dispatcher_send` / `async_dispatcher_connect`.  (I'm aware the camera integration is using the old method)

Reply:
I am looking how Sonos does it since they seem to have something similar where they poll the speakers for data. It sounds like this might be the flow?

* On set up entry, do a `async_dispatcher_connect` for a custom event with `update_tokens` as the callback.
* using the `async_track_time_interval` method I am here, fire the update tokens event on an interval.